It seem to start with this error: https://ci.eclipse.org/jdt/job/eclipse.jdt.core-Github/job/PR-5013/2/consoleText
```
[2026-04-20T09:29:28.577Z] [INFO] --- tycho-apitools:5.0.3-SNAPSHOT:verify (verify) @ org.eclipse.jdt.apt.core ---
[2026-04-20T09:29:28.577Z] [INFO] Resolve API baseline for org.eclipse.jdt:org.eclipse.jdt.apt.core:eclipse-plugin:3.8.700-SNAPSHOT with linux/gtk/x86_64
[2026-04-20T09:29:28.577Z] [INFO] [873ca318-a305-4f09-8f49-a086220624a6][extension>org.eclipse.tycho:tycho-maven-plugin:5.0.3-SNAPSHOT] Error processing mirrors URL: https://www.eclipse.org/downloads/download.php?file=/eclipse/updates/4.39/R-4.39-202602260420&countryCode=us&timeZone=0&format=xml
[2026-04-20T09:29:28.577Z] org.eclipse.core.runtime.CoreException: download from https://www.eclipse.org/downloads/download.php?file=/eclipse/updates/4.39/R-4.39-202602260420&countryCode=us&timeZone=0&format=xml failed
[2026-04-20T09:29:28.578Z]     at org.eclipse.tycho.p2maven.transport.TychoRepositoryTransport.stream (TychoRepositoryTransport.java:180)
[2026-04-20T09:29:28.578Z]     at org.eclipse.equinox.internal.p2.artifact.repository.MirrorSelector.getMirrorsDocument (MirrorSelector.java:319)
[2026-04-20T09:29:28.578Z]     at org.eclipse.equinox.internal.p2.artifact.repository.MirrorSelector.computeMirrors (MirrorSelector.java:270)
[2026-04-20T09:29:28.578Z]     at org.eclipse.equinox.internal.p2.artifact.repository.MirrorSelector.initMirrors (MirrorSelector.java:381)
[2026-04-20T09:29:28.578Z]     at org.eclipse.equinox.internal.p2.artifact.repository.MirrorSelector.selectMirror (MirrorSelector.java:457)
[2026-04-20T09:29:28.578Z]     at org.eclipse.equinox.internal.p2.artifact.repository.MirrorSelector.getMirrorLocation (MirrorSelector.java:356)
[2026-04-20T09:29:28.578Z]     at org.eclipse.equinox.internal.p2.artifact.repository.simple.SimpleArtifactRepository.getMirror (SimpleArtifactRepository.java:792)
[2026-04-20T09:29:28.578Z]     at org.eclipse.equinox.internal.p2.artifact.repository.simple.SimpleArtifactRepository.downloadArtifact (SimpleArtifactRepository.java:675)
[2026-04-20T09:29:28.578Z]     at org.eclipse.equinox.internal.p2.artifact.repository.simple.SimpleArtifactRepository.getArtifact (SimpleArtifactRepository.java:890)
[2026-04-20T09:29:28.578Z]     at org.eclipse.tycho.p2.repository.RepositoryArtifactProvider.getArtifactFromOneMirror (RepositoryArtifactProvider.java:243)
[2026-04-20T09:29:28.578Z]     at org.eclipse.tycho.p2.repository.RepositoryArtifactProvider.getArtifactFromAnyMirror (RepositoryArtifactProvider.java:223)
[2026-04-20T09:29:28.578Z]     at org.eclipse.tycho.p2.repository.RepositoryArtifactProvider$1.perform (RepositoryArtifactProvider.java:198)
[2026-04-20T09:29:28.578Z]     at org.eclipse.equinox.internal.p2.artifact.repository.simple.SimpleArtifactRepository.getArtifact (SimpleArtifactRepository.java:870)
[2026-04-20T09:29:28.578Z]     at org.eclipse.equinox.internal.p2.artifact.repository.simple.SimpleArtifactRepository.getArtifacts (SimpleArtifactRepository.java:959)
[2026-04-20T09:29:28.578Z]     at org.eclipse.tycho.p2.repository.LazyArtifactRepository.getArtifacts (LazyArtifactRepository.java:102)
[2026-04-20T09:29:28.578Z]     at org.eclipse.tycho.p2.repository.RepositoryArtifactProvider.getArtifactFromAnyChildRepository (RepositoryArtifactProvider.java:210)
[2026-04-20T09:29:28.578Z]     at org.eclipse.tycho.p2.repository.RepositoryArtifactProvider.getArtifactFromAnyFormatAvailableInRepository (RepositoryArtifactProvider.java:177)
[2026-04-20T09:29:28.578Z]     at org.eclipse.tycho.p2.repository.RepositoryArtifactProvider.getArtifactFromAnySource (RepositoryArtifactProvider.java:163)
[2026-04-20T09:29:28.578Z]     at org.eclipse.tycho.p2.repository.CompositeArtifactProviderBaseImpl.getArtifact (CompositeArtifactProviderBaseImpl.java:55)
[2026-04-20T09:29:28.578Z]     at org.eclipse.tycho.p2.repository.MirroringArtifactProvider.downloadCanonicalArtifact (MirroringArtifactProvider.java:347)
[2026-04-20T09:29:28.578Z]     at org.eclipse.tycho.p2.repository.MirroringArtifactProvider.downloadMostSpecificNeededFormatOfArtifact (MirroringArtifactProvider.java:340)
[2026-04-20T09:29:28.578Z]     at org.eclipse.tycho.p2.repository.MirroringArtifactProvider.downloadArtifact (MirroringArtifactProvider.java:325)
[2026-04-20T09:29:28.578Z]     at org.eclipse.tycho.p2.repository.MirroringArtifactProvider.makeOneFormatLocallyAvailable (MirroringArtifactProvider.java:264)
[2026-04-20T09:29:28.578Z]     at org.eclipse.tycho.p2.repository.MirroringArtifactProvider.makeLocallyAvailable (MirroringArtifactProvider.java:206)
[2026-04-20T09:29:28.578Z]     at org.eclipse.tycho.p2.repository.MirroringArtifactProvider.getArtifactFile (MirroringArtifactProvider.java:138)
[2026-04-20T09:29:28.578Z]     at org.eclipse.tycho.p2.repository.CompositeArtifactProvider.getArtifactFile (CompositeArtifactProvider.java:89)
[2026-04-20T09:29:28.578Z]     at org.eclipse.tycho.p2resolver.TargetPlatformBaseImpl.getLocalArtifactFile (TargetPlatformBaseImpl.java:169)
[2026-04-20T09:29:28.578Z]     at org.eclipse.tycho.core.resolver.DefaultP2ResolutionResult.lambda$addArtifact$2 (DefaultP2ResolutionResult.java:90)
[2026-04-20T09:29:28.578Z]     at org.eclipse.tycho.core.resolver.DefaultP2ResolutionResultEntry.getLocation (DefaultP2ResolutionResultEntry.java:73)
[2026-04-20T09:29:28.578Z]     at org.eclipse.tycho.osgi.framework.EclipseApplicationFactory.getApiBaselineBundles (EclipseApplicationFactory.java:128)
[2026-04-20T09:29:28.578Z]     at org.eclipse.tycho.apitools.ApiApplicationResolver.getApiBaselineBundles (ApiApplicationResolver.java:57)
[2026-04-20T09:29:28.578Z]     at org.eclipse.tycho.apitools.ApiAnalysisMojo.getBaselineBundles (ApiAnalysisMojo.java:380)
[2026-04-20T09:29:28.578Z]     at org.eclipse.tycho.apitools.ApiAnalysisMojo.execute (ApiAnalysisMojo.java:201)
[2026-04-20T09:29:28.578Z]     at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:126)
[2026-04-20T09:29:28.578Z]     at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:328)
[2026-04-20T09:29:28.578Z]     at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:316)
[2026-04-20T09:29:28.578Z]     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:212)
[2026-04-20T09:29:28.578Z]     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:174)
[2026-04-20T09:29:28.578Z]     at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:75)
[2026-04-20T09:29:28.579Z]     at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:162)
[2026-04-20T09:29:28.579Z]     at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
[2026-04-20T09:29:28.579Z]     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:159)
[2026-04-20T09:29:28.579Z]     at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:105)
[2026-04-20T09:29:28.579Z]     at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:73)
[2026-04-20T09:29:28.579Z]     at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:53)
[2026-04-20T09:29:28.579Z]     at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:118)
[2026-04-20T09:29:28.579Z]     at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:261)
[2026-04-20T09:29:28.579Z]     at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:173)
[2026-04-20T09:29:28.579Z]     at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:101)
[2026-04-20T09:29:28.579Z]     at org.apache.maven.cli.MavenCli.execute (MavenCli.java:906)
[2026-04-20T09:29:28.579Z]     at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:283)
[2026-04-20T09:29:28.579Z]     at org.apache.maven.cli.MavenCli.main (MavenCli.java:206)
[2026-04-20T09:29:28.579Z]     at jdk.internal.reflect.DirectMethodHandleAccessor.invoke (DirectMethodHandleAccessor.java:104)
[2026-04-20T09:29:28.579Z]     at java.lang.reflect.Method.invoke (Method.java:565)
[2026-04-20T09:29:28.579Z]     at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:255)
[2026-04-20T09:29:28.579Z]     at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:201)
[2026-04-20T09:29:28.579Z]     at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:361)
[2026-04-20T09:29:28.579Z]     at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:314)
[2026-04-20T09:29:28.579Z] Caused by: java.io.IOException: Server returned HTTP code: 429 for URL https://www.eclipse.org/downloads/download.php?file=/eclipse/updates/4.39/R-4.39-202602260420&countryCode=us&timeZone=0&format=xml
```

```
[2026-04-20T09:29:28.846Z] Caused by: java.io.IOException: Server returned HTTP code: 429 for URL https://www.eclipse.org/downloads/download.php?file=/eclipse/updates/4.39/R-4.39-202602260420&countryCode=us&timeZone=0&format=xml
```

I can access this url **now**. I saw this morning few error satus mails from eclipse.org status service.

<img width="875" height="252" alt="image" src="https://github.com/user-attachments/assets/d6892f05-1c0d-499f-855a-c981bca08f13" />


See https://www.eclipsestatus.io

I will retrigger the tests manually.

--- a/org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/ast/CaseStatement.java
+++ b/org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/ast/CaseStatement.java
@@ -327,12 +327,11 @@ public void generateCode(BlockScope currentScope, CodeStream codeStream) {
 	int pc = codeStream.position;
 	this.targetLabel.place();
 
-	if (containsPatternVariable(true)) {
+	if (this.constantExpressions.length > 0 && this.constantExpressions[0] instanceof Pattern pattern && (!pattern.isUnguarded() || pattern.containsPatternVariable(true))) {
 
 		BranchLabel patternMatchLabel = new BranchLabel(codeStream);
 		BranchLabel matchFailLabel = new BranchLabel(codeStream);
 
-		Pattern pattern = (Pattern) this.constantExpressions[0];
 		codeStream.load(this.swich.selector);
 		pattern.generateCode(currentScope, codeStream, patternMatchLabel, matchFailLabel);
 		codeStream.goto_(patternMatchLabel);

--- a/org.eclipse.jdt.core.tests.compiler/src/org/eclipse/jdt/core/tests/compiler/regression/SwitchPatternTest.java
+++ b/org.eclipse.jdt.core.tests.compiler/src/org/eclipse/jdt/core/tests/compiler/regression/SwitchPatternTest.java
@@ -9870,4 +9870,82 @@ public class SwitchPatternTest extends AbstractRegressionTest9 {
 			"A null case label has to be either the only expression in a case label or the first expression followed only by a default\n" +
 			"----------\n");
 	}
+
+	// https://github.com/eclipse-jdt/eclipse.jdt.core/issues/5000
+	// Switch guards are not respected when record has no components
+	public void testIssue5000() throws Exception {
+		runConformTest(
+			new String[] {
+				"X.java",
+				"""
+				sealed interface Union {
+					record Foo() implements Union {
+					}
+
+					record Bar() implements Union {
+					}
+				}
+
+				public class X {
+					public static void main(String[] args) {
+						Union foo = new Union.Foo();
+
+						var f = switch (foo) {
+							case Union.Foo() when "".equals("abc") -> {
+								yield "bad";
+							}
+							case Union.Foo() -> {
+								yield "good";
+							}
+							case Union.Bar() -> {
+								yield "bar";
+							}
+						};
+
+						System.out.println(f);
+					}
+				}
+				"""
+			},
+			"good");
+	}
+
+	// https://github.com/eclipse-jdt/eclipse.jdt.core/issues/5000
+	// Switch guards are not respected when record has no components
+	public void testIssue5000_2() throws Exception {
+		runConformTest(
+			new String[] {
+				"X.java",
+				"""
+				sealed interface Union {
+					record Foo() implements Union {
+					}
+
+					record Bar() implements Union {
+					}
+				}
+
+				public class X {
+					public static void main(String[] args) {
+						Union foo = new Union.Foo();
+
+						var f = switch (foo) {
+							case Union.Foo() when "abc".equals("abc") -> {
+								yield "bad";
+							}
+							case Union.Foo() -> {
+								yield "good";
+							}
+							case Union.Bar() -> {
+								yield "bar";
+							}
+						};
+
+						System.out.println(f);
+					}
+				}
+				"""
+			},
+			"bad");
+	}
 }

